modules/objdetect/include/opencv2/objdetect/charuco_detector.hpp

```diff
@@ -0,0 +1,124 @@
+     * @param markerCorners vector of already detected markers corners. For each marker, its four
+     * corners are provided, (e.g std::vector<std::vector<cv::Point2f> > ). For N detected markers, the
+     * dimensions of this array should be Nx4. The order of the corners should be clockwise.
+     * If markerCorners and markerCorners are empty, the detectMarkers() will run and detect aruco markers and ids
+     * @param markerIds list of identifiers for each marker in corners.
+     * If markerCorners and markerCorners are empty, the detectMarkers() will run and detect aruco markers and ids
+     * @param charucoCorners interpolated chessboard corners
+     * @param charucoIds interpolated chessboard corners identifiers
+     *
+     * This function receives the detected markers and returns the 2D position of the chessboard corners
+     * from a ChArUco board using the detected Aruco markers.
+     *
+     * If markerCorners and markerCorners are empty, the detectMarkers() will run and detect aruco markers and ids
+     *
+     * If camera parameters are provided, the process is based in an approximated pose estimation, else it is based on local homography.
+     * Only visible corners are returned. For each corner, its corresponding identifier is
+     * also returned in charucoIds.
+     */
+    CV_WRAP void detectBoard(InputArray image, InputOutputArrayOfArrays markerCorners, InputOutputArray markerIds,
+                             OutputArray charucoCorners, OutputArray charucoIds) const;
+
+    /**
+     * @brief Detect ChArUco Diamond markers
+     *
+     * @param image input image necessary for corner subpixel.
+     * @param markerCorners list of detected marker corners from detectMarkers function.
+     * If markerCorners and markerCorners are empty, the detectDiamonds() will run and detect aruco markers and ids
+     * @param markerIds list of marker ids in markerCorners.
+     * If markerCorners and markerCorners are empty, the detectDiamonds() will run and detect aruco markers and ids
+     * @param diamondCorners output list of detected diamond corners (4 corners per diamond). The order
+     * is the same than in marker corners: top left, top right, bottom right and bottom left. Similar
+     * format than the corners returned by detectMarkers (e.g std::vector<std::vector<cv::Point2f> > ).
+     * @param diamondIds ids of the diamonds in diamondCorners. The id of each diamond is in fact of
+     * type Vec4i, so each diamond has 4 ids, which are the ids of the aruco markers composing the
+     * diamond.
+     *
+     * This function detects Diamond markers from the previous detected ArUco markers. The diamonds
+     * are returned in the diamondCorners and diamondIds parameters. If camera calibration parameters
+     * are provided, the diamond search is based on reprojection. If not, diamond search is based on
+     * homography. Homography is faster than reprojection but can slightly reduce the detection rate.
```

Comment:
"Homography is faster than reprojection, but less accurate."

Reply:
fixed